app/models/middleware_server_performance.rb

```diff
@@ -16,7 +16,11 @@ def self.build_results_for_report_middleware(options)
     start_time, end_time, interval = parse_time_interval(options)
     results = []
     MiddlewareServer.find_each do |ms|
-      raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
+      begin
+        raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
```

Comment:
@abonas it's usually good for the library to have a generic connection error exception that THIS code can rescue.  That exception can have subclasses for each of the types you need and the generic one can be used for this rescue.  The problem with rescuing StandardError here (via the default) is you rescue exceptions you don't really want to rescue here.  This makes it less obvious that you are skipping only connection related errors.

For example, see [faraday's error.rb](https://github.com/lostisland/faraday/blob/09cdff22e0954beb45dd26158275e8842a5746d3/lib/faraday/error.rb#L46), within the Faraday namespace is a ClientError and ConnectionFailed subclass of that error.

Does that make sense?